pkg/apis/core/validation/validation.go

```diff
@@ -3550,15 +3550,37 @@ func validatePodAffinity(podAffinity *core.PodAffinity, fldPath *field.Path) fie
 	return allErrs
 }
 
+func ValidateSeccompProfileField(sp core.SeccompProfile, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if err := ValidateSeccompProfileType(fldPath.Child("type"), sp.Type); err != nil {
+		allErrs = append(allErrs, err)
+	}
+
```

Comment:
restructure to avoid repeated checks of the same conditions:

```go
if sp.Type == core.SeccompProfileLocalhost {
  if sp.LocalhostProfile == nil {
    allErrs = append(allErrs, field.Required(fldPath.Child("localhostProfile"), ...
  } else {
    allErrs = append(allErrs, validateLocalDescendingPath(...
  }
} else {
  if sp.LocalhostProfile != nil {
    allErrs = append(allErrs, field.Invalid(fldPath.Child("localhostProfile"), ...
  }
}
```

Reply:
Amended the code with your suggestion.